smartsim/_core/_cli/cli.py

```diff
@@ -73,14 +74,34 @@ def execute(self, cli_args: t.List[str]) -> int:
             self.parser.print_help()
             return 0
 
-        app_args = cli_args[1:]
-        self.args = self.parser.parse_args(app_args)
+        app_args = cli_args[1:]  # exclude the path to executable
+        subcommand = cli_args[1]  # first positional arg is the subcommand
 
-        if not (menu_item := self.menu.get(app_args[0], None)):
+        menu_item = self.menu.get(subcommand, None)
+        if not menu_item:
             self.parser.print_help()
             return 0
```

Comment:
Not part of this PR, but noticed it and wanted your opinion while we are here: I'm not sure that I like that if a user requests a sub-command that does not exist, a successful return code is given back. Thoughts on making this a non-zero return code (I think `argparse` uses 2 by default, but I would need to check)

Reply:
If we're automating calls to the CLI, I suppose we can't tell that it failed to do something because it gave back a zero when given a bad arg.... That seals it for me, tbh.

I agree that it should be non-zero. I'll give back 2 to be consistent w/argparse. Great catch.